bibliopixel/layout/layout.py

```diff
@@ -92,31 +93,35 @@ def set_colors(self, buf):
     def setBuffer(self, buf):
         """DEPRECATED!"""
         # https://stackoverflow.com/questions/1624883
-        self.set_colors(buf=list(zip(*(iter(buf),) * 3)))
+        self.set_colors(buf=list(zip(*(iter(buf),) * self.pixel_size)))
 
     def set_brightness(self, brightness):
         self.brightness = brightness
         for d in self.drivers:
             d.set_brightness(brightness)
 
     # Set single pixel to RGB value
-    def setRGB(self, pixel, r, g, b):
+    def setRGB(self, pixel, *color):
```

Comment:
Personally, all reference to RGB needs to just go away (with backward-compat overrides of course).